src/util/records.py

```diff
@@ -310,6 +311,69 @@ def remove_asset(cr, name):
 # fmt:on
 
 
+def remove_action(cr, xml_id=None, action_id=None):
+    assert bool(xml_id) ^ bool(action_id)
+
+    if xml_id:
+        action_id = ref(cr, xml_id)
+
+    action_model = None
+    if action_id:
+        module, _, name = xml_id.partition(".")
```

Comment:
This cannot work when the xmlid is not provided.

You should get the model from the `type` column of `ir_actions` table.